test-suites/_pool_core_erc721_auction_liquidation.spec.ts

```diff
@@ -0,0 +1,918 @@
+        await pool
+          .connect(liquidator.signer)
+          .startAuction(borrower.address, bayc.address, 0)
+      );
+
+      await changePriceAndValidate(bayc, "101");
+
+      // end auction
+      await pool
+        .connect(borrower.signer)
+        .setAuctionValidityTime(borrower.address);
+
+      await changePriceAndValidate(bayc, "8");
+
+      // try to liquidate the NFT,failure of liquidation
+      await expect(
+        pool
+          .connect(liquidator.signer)
+          .liquidationERC721(
+            bayc.address,
+            borrower.address,
+            0,
+            parseEther("24").toString(),
+            false,
+            {gasLimit: 5000000}
+          )
+      ).revertedWith(ProtocolErrors.AUCTION_NOT_STARTED);
+      expect(await nBAYC.balanceOf(borrower.address)).to.be.equal(3);
+    });
+
+    it("TC-auction-liquidation-28 If ERC721 HF>=1.5, run Update AuctionValidityTime but not EndAuction. When ERC721 HF<1, StartAuction is started again and this ERC721 can be liquidated", async () => {
+      const {
+        users: [borrower, liquidator],
+        pool,
+        bayc,
+        nBAYC,
+      } = testEnv;
+
+      // drop BAYC price to liquidation levels (HF = 0.6)
+      await changePriceAndValidate(bayc, "8");
+
+      // start auction
+      await waitForTx(
+        await pool
+          .connect(liquidator.signer)
+          .startAuction(borrower.address, bayc.address, 0)
+      );
+
+      await changePriceAndValidate(bayc, "101");
+
+      // end auction
+      await pool
+        .connect(borrower.signer)
+        .setAuctionValidityTime(borrower.address);
+
+      await changePriceAndValidate(bayc, "8");
+
+      // start auction
+      await waitForTx(
+        await pool
+          .connect(liquidator.signer)
+          .startAuction(borrower.address, bayc.address, 0)
+      );
+      // try to liquidate the NFT
+      await pool
+        .connect(liquidator.signer)
+        .liquidationERC721(
+          bayc.address,
+          borrower.address,
+          0,
+          parseEther("24").toString(),
+          false,
+          {gasLimit: 5000000}
+        );
+      expect(await nBAYC.balanceOf(borrower.address)).to.be.lt(3);
+    });
+
+    it("TC-auction-liquidation-29 Liquidator tries to start auction again (should be reverted)", async () => {
+      const {
+        users: [borrower, liquidator],
+        pool,
+        bayc,
+      } = testEnv;
+
+      await changePriceAndValidate(bayc, "8");
+
+      // start auction
+      await waitForTx(
+        await pool
+          .connect(liquidator.signer)
+          .startAuction(borrower.address, bayc.address, 0)
+      );
+
+      // start auction again
+      await expect(
+        pool
+          .connect(liquidator.signer)
+          .startAuction(borrower.address, bayc.address, 0)
+      ).to.be.revertedWith(ProtocolErrors.AUCTION_ALREADY_STARTED);
+    });
+  });
+
+  describe("Do not revert to snapshot on every step", () => {
```

Comment:
suggest we merge this with revert and keep every use-case atomic and independent with each other